iothub_client/src/iothubtransport_mqtt_common.c

```diff
@@ -236,7 +240,7 @@ typedef struct MQTTTRANSPORT_HANDLE_DATA_TAG
     int http_proxy_port;
     char* http_proxy_username;
     char* http_proxy_password;
-    bool isProductInfoSet;
+    bool isConnectUsernameSet;
```

Comment:
Is azure sphere going to complain about this?

Reply:
;tl-dr - NO

Since you asked...

We still will send productInfo if set by application.  See `buildConfigForUsernameStep2IfNeeded`.  I changed the name of the variable because it's now not just productInfoSet that matters, but generally adding to userName (product & modelID).

There is a *slight* behavior change here.  Before if the app did below:

```c
IoThub_CreateHandleFromConnectionString();
IoTHub_SendTelemetry(h);
// wait for a bunch, at least long enough for CONNECT to go through
// And after time passed
IoTHub_SetOption("product_info", p);
```

Then if there were ever a disconnection and a successful re-connection, then the product_info would be appending to the subsequent CONNECT packet.

In the new world, this product_info would not be appending to subsequent CONNECT packet because we only build up userName once.  Since an app doesn't control reconnections and since setting product info so late would be odd, I don't see this change as being problematic.  Trying to maintain old behavior is going to really complicate things also would demand specific tests for permutations of the app I called out above to make sure productId was set (or not).